src/librustc_mir/transform/lower_128bit.rs

```diff
@@ -23,14 +24,30 @@ pub struct Lower128Bit;
 impl MirPass for Lower128Bit {
     fn run_pass<'a, 'tcx>(&self,
                           tcx: TyCtxt<'a, 'tcx, 'tcx>,
-                          _src: MirSource,
+                          source: MirSource,
                           mir: &mut Mir<'tcx>) {
         let debugging_override = tcx.sess.opts.debugging_opts.lower_128bit_ops;
         let target_default = tcx.sess.host.options.i128_lowering;
         if !debugging_override.unwrap_or(target_default) {
             return
         }
 
+        // Don't do the lowering in const contexts.  It's not needed, and
+        // trying to do so causes ICEs because the MIR for the lang items
+        // this calls isn't available to evaluate.
+        let id = tcx.hir.as_local_node_id(source.def_id).unwrap();
+        match (tcx.hir.body_owner_kind(id), source.promoted) {
+            (_, Some(_)) |
+            (hir::BodyOwnerKind::Const, _) |
+            (hir::BodyOwnerKind::Static(_), _) => return,
+
+            (hir::BodyOwnerKind::Fn, _) => {
+                if tcx.is_const_fn(source.def_id) {
```

Comment:
Does this mean i128 ops won't get lowered in const fns *at all*? As const fns can run at runtime as well, I figure they'd have to be lowered for the run time version even if const eval can handle them.